#2384 Added test script in Meta Data Source on new UI

Patrykb0802 opened the pull request:
In point.vue:
- Added validateScript() method
- Added 'Run script' button to  Meta Data Point editing
- Added alert that shows result of working script
- Modified script textarea to run validateScript method
- Added new rule that is checking if script is correct

- Added script.runScript translation key to en.json

Patrykb0802 committed:
#2384 Added test script in Meta Data Source on new UI

In point.vue:
- Added validateScript() method
- Added 'Run script' button to  Meta Data Point editing
- Added alert that shows result of working script
- Modified script textarea to run validateScript method
- Added new rule that is checking if script is correct

- Added script.runScript translation key to en.json

Limraj commented on scadalts-ui/src/components/datasources/MetaDataSource/point.vue:
```diff
@@ -317,14 +317,26 @@
 								v-model="datapoint.pointLocator.script"
 								rows=3
+                @input="validateScript"
+                :rules="[ruleValidScript, ruleNotNull]"
```
The file formatting needs to be corrected.

Limraj reviewed (commented):
Problem 1
1. Click on 'Create new data point';
2. I click on the script field and then click on another place in this window -> the field with the script turns red without an error message that this field cannot be empty;

Expected behavior:
If the field is empty, then in addition to marking it in red, a message should appear saying that the field cannot be empty;

Problem 2
1. Click on 'Create new data point';
2. I click on the script field and then enter "return" -> validation requests are sent every letter entered, i.e. first validate with r, then re, then ret etc.

Expected behavior:
Validation should be activated if we stop editing this window, i.e. we click on another fields;

Problem 3
1. Click on 'Create new data point';
2. I click on 'Run scirpt' on an empty field -> a green message appears and result false;

Expected behavior:
If the field is empty, the run script should not succeed;

Limraj commented on scadalts-ui/src/components/datasources/MetaDataSource/point.vue:
```diff
@@ -535,6 +550,21 @@ export default {
+        this.validScript = resp.success;
+        this.resultMessage = resp.message;
+        this.$refs.scriptBodyTextarea.validate();
+        this.$t(resp);
```
'this.$t(resp);' should probably be removed.

Patrykb0802 committed:
#2384 Added test script in Meta Data Source on new UI

- Added correct error message when creating new data point and script field is empty

- Added delayed validation, so requests for validation are not send every letter typed

- Added handling of error when field for script is empty and you press "run script", now when field is empty and you press "run script" nothing happens

- Deleted duplicated property key dox.newUISettings

Limraj commented on scadalts-ui/src/components/datasources/MetaDataSource/point.vue:
```diff
@@ -317,14 +317,29 @@
+                  @focus="validateScript"
+                  @input="delayedValidateScript"
+                  :rules="[ruleNotNull]"
+                  ref="scriptBodyTextarea"
```
You need to improve the formatting of the code so that the changes themselves are visible and that the code is formatted correctly.

Patrykb0802 committed:
#2384 Added test script in Meta Data Source on new UI
